dtfs-central-api/src/repositories/utilisation-reports-repo/utilisation-report-sql.repo.ts

```diff
@@ -1,10 +1,35 @@
 // TODO FN-1853 - rename this to `utilisation-report.repo.ts` when all repo
 //  methods have been migrated from MongoDB to SQL
 import { SqlDbDataSource } from '@ukef/dtfs2-common/sql-db-connection';
-import { UtilisationReportEntity, ReportPeriod } from '@ukef/dtfs2-common';
+import { UtilisationReportEntity, ReportPeriod, UTILISATION_REPORT_RECONCILIATION_STATUS } from '@ukef/dtfs2-common';
+
+export type GetUtilisationReportDetailsOptions = {
+  reportPeriod?: ReportPeriod;
+  excludeNotUploaded?: boolean;
+};
 
 export const UtilisationReportRepo = SqlDbDataSource.getRepository(UtilisationReportEntity).extend({
   async findOneByBankIdAndReportPeriod(bankId: string, reportPeriod: ReportPeriod): Promise<UtilisationReportEntity | null> {
     return await this.findOneBy({ bankId, reportPeriod });
   },
+
+  async findAllByBankId(bankId: string, options?: GetUtilisationReportDetailsOptions): Promise<UtilisationReportEntity[]> {
+    let query = this.createQueryBuilder('report').where('report.bankId = :bankId', { bankId });
+    if (options?.excludeNotUploaded) {
+      query = query
+        .leftJoinAndSelect('report.azureFileInfo', 'azureFileInfo')
+        .andWhere('azureFileInfo.utilisationReportId IS NOT NULL')
+        .andWhere('report.status <> :status', { status: UTILISATION_REPORT_RECONCILIATION_STATUS.REPORT_NOT_RECEIVED });
+    }
+
+    if (options?.reportPeriod) {
+      query = query
+        .andWhere('report.reportPeriodStartMonth = :startMonth', { startMonth: options.reportPeriod.start.month })
+        .andWhere('report.reportPeriodEndMonth = :endMonth', { endMonth: options.reportPeriod.end.month })
+        .andWhere('report.reportPeriodStartYear = :startYear', { startYear: options.reportPeriod.start.year })
+        .andWhere('report.reportPeriodEndYear = :endYear', { endYear: options.reportPeriod.end.year });
```

Comment:
`reportPeriodStartMonth` etc are the actual db column names, which are generated by typeORM from the nested structure we have in our entity type. I don't think there is a way around using them explicitly but it did feel a bit funky to do so, similarly with using the `azureFileInfo.utilisationReportId` above. I'm thinking this might just be what it is like working with the query builder and I perhaps just need to get used to it, but just wanting to highlight in case I am missing any tricks here!